### ISSUE TITLE
Optimize plotting performance by drawing only visible regions

### ISSUE THREAD

**Issue by olegtarasov** [oxyplot/oxyplot#834]
### Feature description

Right now most series iterate through all data points and try to draw them with no regard to what region is actually visible. The example of an optimized is CandleStickSeries, where only visible data points are being drawn. The benefit of such an optimization is huge, because rendering time can be made independent of the number of data points.
I propose to extract the code that deals with visible window coordinates from CandleStickSeries, make a generic version of it in XYAxisSeries and optimize performance of different series.

**Comment by objorke** [comment]
Most series do not have a requirement that the data should be monotonic, so I think you need to iterate through all data to check visibility.

**Comment by olegtarasov** [comment]
That's true, but in many cases the data is actually monotonic, so we can use this optimization. We can add a simple check to UpdateData() method, where we could detect monotonicity, and if found, use the rendering optimization.

**Comment by objorke** [comment]
that's a good idea! it increases the complexity a bit, but it is probably worth it.

**Comment by olegtarasov** [comment]
I've added monotonicity checks and pull request #833 is now ready.

**Comment by olegtarasov** [comment]
In pull request #840 I optimized rendering performance in RectangleBarSeries and added tracker optimization for AreaSeries. I checked my changes against StyleCop rules this time.
